tests/bdd/test_volume_create.py

```diff
@@ -0,0 +1,195 @@
+from openapi.openapi_client.model.volume_status import VolumeStatus
+
+VOLUME_UUID = "5cd5378e-3f05-47f1-a830-a0f5873a1449"
+VOLUME_SIZE = 10485761
+NUM_VOLUME_REPLICAS = 1
+REST_SERVER = "http://localhost:8081/v0"
+CREATE_REQUEST_KEY = "create_request"
+POOL_UUID = "4cc6ee64-7232-497d-a26f-38284a444980"
+NODE_NAME = "mayastor"
+
+
+# This fixture will be automatically used by all tests.
+# It starts the deployer which launches all the necessary containers.
+# A pool is created for convenience such that it is available for use by the tests.
+@pytest.fixture(autouse=True)
+def init():
+    common.deployer_start(1)
+    common.get_pools_api().put_node_pool(
+        NODE_NAME, POOL_UUID, CreatePoolBody(["malloc:///disk?size_mb=500"])
+    )
+    yield
+    common.deployer_stop()
+
+
+# Fixture used to pass the volume create request between test steps.
+@pytest.fixture(scope="function")
+def create_request():
+    return {}
+
+
+@scenario(
+    "features/volume/create.feature", "provisioning failure due to missing Mayastor"
+)
+def test_provisioning_failure():
+    """provisioning failure."""
+
+
+@scenario(
+    "features/volume/create.feature", "desired number of replicas cannot be created"
+)
+def test_desired_number_of_replicas_cannot_be_created():
+    """desired number of replicas cannot be created."""
+
+
+@scenario("features/volume/create.feature", "sufficient suitable pools")
+def test_sufficient_suitable_pools():
+    """sufficient suitable pools."""
+
+
+@given("a control plane, Mayastor instances and a pool")
+def a_control_plane_a_mayastor_instance_and_a_pool():
+    """a control plane, Mayastor instances and a pool."""
+    docker_client = docker.from_env()
+
+    # The control plane comprises the core agents, rest server and etcd instance.
+    for component in ["core", "rest", "etcd"]:
+        try:
+            container = docker_client.containers.get(component)
+        except docker.errors.NotFound as exc:
+            raise Exception("{} container not found", component)
+        else:
+            container_state = container.attrs["State"]
+            if container_state["Status"] != "running":
+                raise Exception("{} container not running", component)
+
+    # Check all Mayastor instances are running
+    try:
+        mayastors = docker_client.containers.list(
+            all=True, filters={"name": "mayastor"}
+        )
+    except docker.errors.NotFound:
+        raise Exception("No Mayastor instances")
+
+    for mayastor in mayastors:
+        container_state = mayastor.attrs["State"]
+        if container_state["Status"] != "running":
+            raise Exception("{} container not running", mayastor.attrs["Name"])
+
+    # Check for a pool
+    pool = common.get_pools_api().get_pool(POOL_UUID)
+    assert pool.id == POOL_UUID
+
+
+@given("a request for a volume")
+def a_request_for_a_volume(create_request):
+    """a request for a volume."""
+    policy = {"self_heal": False, "topology": None}
```

Comment:
Ah I had forgotten about policies, I don't think the reconciler is respecting the `self_heal`! :(